tox.ini

```diff
@@ -62,13 +66,15 @@ commands =
 [pinned]
 deps =
     -ctests/constraints.txt
-    cryptography==2.0
+    #using cryptography-2.3.1 to solve dependencies conflict on pip20.2>=
+    cryptography==2.3.1
     cssselect==0.9.1
     itemadapter==0.1.0
     parsel==1.5.0
     Protego==0.1.15
     PyDispatcher==2.0.5
```

Comment:
Looking at https://github.com/scrapy/scrapy/issues/1773#issuecomment-327620131 , it looks like we may need to upgrade `PyDispatcher` to `>=2.1.0` on PyPy.

So, assuming that version upgrade fixes the issue in the first of the 2 PyPy jobs, I would suggest:

1. Updating `setup.py` to require `PyDispatcher>=2.1.0` when using PyPy.
2. In `tox.ini`, creating a `[testenv:pypy-pinned]` section similar to the Windows one, and following a similar approach to install `PyDispatcher==2.1.0` only in the `pypy-pinned` environment.
3. In `travis.yml`, switch the first PyPy job from `pinned` to `pypy-pinned`.

Reply:
I think [#1773 comment](https://github.com/scrapy/scrapy/issues/1773#issuecomment-327620131) mentions `PyPyDispatcher >= 2.1.0` instead of `PyDispatcher`. (which is in pypy-pinned deps and setup.py)